OAuth2/Infrastructure/SafeExtensions.cs

```diff
@@ -14,5 +15,13 @@ public static class SafeExtensions
         {
             return instance == null ? default(T) : selector(instance);
         }
+
+        /// <summary>
+        /// Executes selector on instance and returns result or returns default value of target type if given instance is null.
+        /// </summary>
+        public static async Task<T> SafeGetAsync<TModel, T>(this TModel instance, Func<TModel, Task<T>> selector) where TModel : class
+        {
+            return instance == null ? default(T) : await selector(instance);
```

Comment:
Is there any reason this is here? Also this should have a .ConfigureAwait(false);